pkg/api/validation/validation.go

```diff
@@ -522,6 +527,22 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateFlocker(flocker *api.FlockerVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if flocker.DatasetName == "" {
+		allErrs = append(allErrs, errs.NewFieldRequired("datasetName"))
+	}
+	if strings.Contains(flocker.DatasetName, "/") {
+		allErrs = append(allErrs, errs.NewFieldInvalid("datasetName", flocker.DatasetName, "must not contain '/'"))
+	}
+	if flocker.Size != "" {
+		if _, err := units.FromHumanSize(flocker.Size); err != nil {
```

Comment:
we have a type we use in our API for human-readable quantities - `resource.Quantity` - if you make the API field be a quanitity then you can validate it without this 'units' library, and it supports common syntax with our other resource variants.